src/Analyzer/QueryTreePassManager.cpp

```diff
@@ -264,6 +264,9 @@ void addQueryTreePasses(QueryTreePassManager & manager)
     manager.addPass(std::make_unique<RewriteArrayExistsToHasPass>());
     manager.addPass(std::make_unique<NormalizeCountVariantsPass>());
 
+    /// should before AggregateFunctionsArithmericOperationsPass
```

Comment:
If it should be applied/not applied depends on the `optimize_aggregators_of_group_by_keys` setting

Reply:
It is checked here https://github.com/ClickHouse/ClickHouse/blob/cadaa7bf81affdecfb0614a374f53fcbdbd5db99/src/Analyzer/Passes/AggregateFunctionOfGroupByKeysPass.cpp#L35-L36